src/login/logind-dbus.c

```diff
@@ -890,7 +915,7 @@ static int method_create_session(sd_bus_message *message, void *userdata, sd_bus
                 goto fail;
 
         session_set_user(session, user);
-        r = session_set_leader(session, leader);
+        r = session_set_leader(session, TAKE_PIDREF(leader));
```

Comment:
so here you donate the PidRef to session_set_leader() even on error, but the function is not ready for that.

Reply:
Thank you! This is good catch indeed!